File: Patterns/Manuscript/manuscript/code/python/abstractfactory/ParkVisitor.py
from AdventureLandFactory import AdventureLandFactory
from FantasyLandFactory import FantasyLandFactory

if __name__ == "__main__":
    # Choose themed area (factory)
    adventureLandFactory = AdventureLandFactory()
    fantasyLandFactory = FantasyLandFactory()

    # Order themed food items
    tropicalIceCream = adventureLandFactory.createIceCream()
    exoticPizza = adventureLandFactory.createPizza()

    magicalIceCream = fantasyLandFactory.createIceCream()
    enchantedPizza = fantasyLandFactory.createPizza()

    # Enjoy themed food items
    tropicalIceCream.enjoy()
    exoticPizza.enjoy()
    magicalIceCream.enjoy()
    enchantedPizza.enjoy()

File: code/csharp/DesignPatterns/abstractfactory/ParkVisitor.cs
namespace AbstractFactory;

public class ParkVisitor
{
    public static void VisitPark()
    {
        // Choose themed area (factory)
        IAbstractFoodStallFactory adventureLandFactory =
                                                    new AdventureLandFactory();
        IAbstractFoodStallFactory fantasyLandFactory = new FantasyLandFactory();

        // Order themed food items
        IIceCream tropicalIceCream = adventureLandFactory.CreateIceCream();
        Pizza exoticPizza = adventureLandFactory.CreatePizza();

        IIceCream magicalIceCream = fantasyLandFactory.CreateIceCream();
        Pizza enchantedPizza = fantasyLandFactory.CreatePizza();

        // Enjoy themed food items
        tropicalIceCream.Enjoy();
        exoticPizza.Enjoy();
        magicalIceCream.Enjoy();
        enchantedPizza.Enjoy();
    }
}
